src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -952,6 +956,25 @@
                                             [server-id (initialize-context)]))
                  :default-server (get-default-server-from-config config)))
 
+(schema/defn ^:always-validate reload-crl-on-change!
+  "Reload the CRL file used by the supplied ssl-context-factory whenever any
+  changes to the file occur."
+  [ssl-context-factory :- InternalSslContextFactory
+   watcher :- (schema/protocol watch-protocol/Watcher)]
+  (when-let [crl-path (.getCrlPath ssl-context-factory)]
+    (let [normalized-crl-path (.getCanonicalPath (fs/file crl-path))]
+      (watch-protocol/add-watch-dir! watcher (fs/parent crl-path)
```

Comment:
adding a flag to the add-watch-dir! service function to disable recursing seems like it wouldn't be a bad idea to me?